packages/engine/paima-rest/package.json

```diff
@@ -7,9 +7,9 @@
   "types": "build/index.d.ts",
   "scripts": {
     "lint:eslint": "eslint .",
-    "build": "tsc --build tsconfig.build.json",
-    "prebuild": "npm run compile:api",
-    "compile:api": "npx tsoa spec-and-routes"
+    "build": "npm run compile:api && tsc --build tsconfig.build.json",
```

Comment:
What I ran into is that if this stays in prebuild, tsoa can't import the PRC-1 types from `utils-backend` on clean builds, or otherwise sees the old definitions, because they're produced on that package's build.

Maybe can somehow be improved by teaching NX to build utils-backend before prebuilding paima-rest?

Reply:
NX should be able to figure out dependencies based on the `references` inside the tsconfig IIRC

Reply:
Seems to not work. It doesn't infer that rest:prebuild depends on utils-backend:build

```
~/paima-engine (patch/standard-achievements-api|✚3) $ rm -r packages/node-sdk/paima-utils-backend/build/
~/paima-engine (patch/standard-achievements-api|✚3) $ npm run build

> @paima/root@0.0.0 prebuild
> npx nx run-many --parallel=${NX_PARALLEL:-3} -t prebuild

 
    ✔  nx run @paima/utils:prebuild (2s)

    ✖  nx run @paima/rest:prebuild
       > @paima/rest@1.0.0 prebuild
       > npm run compile:api
       
       
       > @paima/rest@1.0.0 compile:api
       > npx tsoa spec-and-routes
       
       Generate routes error.
        GenerateMetadataError: No declarations found for referenced type AchievementPublicList.
           at TypeResolver.getModelTypeDeclarations (/home/paima/paima-engine/node_modules/@tsoa/cli/dist/metadataGeneration/typeResolver.js:1125:19)
           at TypeResolver.calcRefTypeName (/home/paima/paima-engine/node_modules/@tsoa/cli/dist/metadataGeneration/typeResolver.js:685:39)
           at TypeResolver.calcTypeReferenceTypeName (/home/paima/paima-engine/node_modules/@tsoa/cli/dist/metadataGeneration/typeResolver.js:876:34)
           at TypeResolver.getReferenceType (/home/paima/paima-engine/node_modules/@tsoa/cli/dist/metadataGeneration/typeResolver.js:886:27)
           at TypeResolver.resolve (/home/paima/paima-engine/node_modules/@tsoa/cli/dist/metadataGeneration/typeResolver.js:513:36)
           at TypeResolver.resolve (/home/paima/paima-engine/node_modules/@tsoa/cli/dist/metadataGeneration/typeResolver.js:503:118)
           at MethodGenerator.Generate (/home/paima/paima-engine/node_modules/@tsoa/cli/dist/metadataGeneration/methodGenerator.js:62:78)
           at /home/paima/paima-engine/node_modules/@tsoa/cli/dist/metadataGeneration/controllerGenerator.js:46:41
           at Array.map (<anonymous>)
           at ControllerGenerator.buildMethods (/home/paima/paima-engine/node_modules/@tsoa/cli/dist/metadataGeneration/controllerGenerator.js:46:14)
       npm ERR! Lifecycle script `compile:api` failed with error: 
       npm ERR! Error: command failed 
       npm ERR!   in workspace: @paima/rest@1.0.0 
       npm ERR!   at location: /home/paima/paima-engine/packages/engine/paima-rest 
       npm ERR! Lifecycle script `prebuild` failed with error: 
       npm ERR! Error: command failed 
       npm ERR!   in workspace: @paima/rest@1.0.0 
       npm ERR!   at location: /home/paima/paima-engine/packages/engine/paima-rest 
       
       

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target prebuild for 2 projects (4s)
 
    ✔    1/2 succeeded [0 read from cache]
 
    ✖    1/2 targets failed, including the following:
         - nx run @paima/rest:prebuild
 ```